### PR TITLE
Improve error messages for Avro/SR interaction. 

### LINKED ISSUE #707: KSQL should include expected Schema name in Schema Registry error
Per [Community Slack discussion](https://confluentcommunity.slack.com/archives/C6UJNMY67/p1518035685000101?thread_ts=1517844132.000663&cid=C6UJNMY67), currently KSQL throws an error if the Avro schema cannot be found in the Schema Registry: 

     Could not fetch the AVRO schema from schema registry. Subject not found.; error code: 40401 

A more user-friendly error would be to include the name of the schema that's expected. For example: 

     Could not fetch the Avro schema "PerFrameStats-value" from schema registry. Subject not found.; error code: 40401

### PR DESCRIPTION
**Pull request by rmoff**
Fixes #707, #1555, helps with #713

### Description 

Existing error message is ambiguous and too generic. Has led to lots of confusion in the community. This is intended to address that. 

Two error conditions are handled, with the current/new messages as below: 

##### Current

```
ksql> create stream twitter with (kafka_topic='twitter_avro_03',value_format='avro');
 Unable to verify the AVRO schema is compatible with KSQL. Map key must be of type STRING
```

##### New

```
ksql> create stream twitter with (kafka_topic='twitter_avro_03',value_format='avro');
KSQL does not currently support the Avro schema for topic twitter_avro_03.
Reason: Map key must be of type STRING

Please see https://github.com/confluentinc/ksql/issues/ to see if this particular reason is already
known, and if not log a new issue. Please include the full Avro schema that you are trying to use.
```

##### Current

```
ksql> create stream wh with (kafka_topic='warehouse_location2', value_format='avro');
 Unable to verify the AVRO schema is compatible with KSQL. Subject not found.; error code: 40401
```

##### New

```
ksql> create stream wh with (kafka_topic='warehouse_location', value_format='avro');
Avro schema for message values on topic warehouse_location does not exist in the Schema Registry.
Subject: warehouse_location-value

Possible causes include:
- The topic itself does not exist
        -> Use SHOW TOPICS; to check
- Messages on the topic are not Avro serialized
        -> Use PRINT 'warehouse_location' FROM BEGINNING; to verify
- Messages on the topic have not been serialized using the Confluent Schema Registry Avro serializer
        -> See https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html
- The schema is registered on a different instance of the Schema Registry
        -> Use the REST API to list available subjects
           https://docs.confluent.io/current/schema-registry/docs/api.html#get--subjects
```

### Testing done 

Updated relevant tests in `AvroUtilTest`.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java [hunk 1]
```diff
@@ -64,34 +65,64 @@ public static AbstractStreamCreateStatement checkAndSetAvroSchema(
...
-        return abstractStreamCreateStatementCopy;
-      } else {
-        return abstractStreamCreateStatement;
-      }
-    } catch (Exception e) {
-      String errorMessage = String.format(
-          " Unable to verify the AVRO schema is compatible with KSQL. %s ",
-          e.getMessage()
+    } catch (final Exception e) {
+      throw new KsqlException("KSQL does not currently support the Avro schema for topic "+kafkaTopicName+
```
Comment (rodesai) [inline ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java]: There may be different kinds of errors thrown here (e.g. some sort of network error). Also, with this PR (https://github.com/confluentinc/ksql/pull/1529) we wouldn't fail with this error. Instead, types that aren't supported will be omitted from the ksql schema. So maybe keep this one as is?

File: ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java [hunk 2]
```diff
@@ -64,34 +65,64 @@ public static AbstractStreamCreateStatement checkAndSetAvroSchema(
...
+        throw new KsqlException("Avro schema for message values on topic " + kafkaTopicName +
+                " does not exist in the Schema Registry.\n"+
+                "Subject: " + kafkaTopicName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX + "\n\n" +
+                "Possible causes include:\n" +
+                "- The topic itself does not exist\n"+
+                "\t-> Use SHOW TOPICS; to check\n" +
+                "- Messages on the topic are not Avro serialized\n" +
+                "\t-> Use PRINT '"+ kafkaTopicName+"' FROM BEGINNING; to verify\n" +
+                "- Messages on the topic have not been serialized using the Confluent Schema Registry Avro serializer\n" +
+                "\t-> See https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html\n" +
```
Comment (rodesai) [inline ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java]: Can we make the links in the error messages constants accessible from the whole codebase? That way if we need to change the links for whatever reason we can do so easily.

Comment (rmoff) [reply ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java]: That's beyond the scope of what I can do in this PR. Is it a blocker?

Comment (rodesai) [reply ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java]: It should just be a matter of adding a static member to https://github.com/confluentinc/ksql/blob/master/ksql-common/src/main/java/io/confluent/ksql/util/KsqlConstants.java and then referencing it here.

Comment (rmoff) [reply ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java]: Done :)

File: ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java (hunk 1 above)
Comment (rmoff) [reply ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java]: I've reworded it

**Pull request merged**